git-webkit is not posting PR links to bugzilla bugs

briannafan opened the pull request:
#### bdfd10f2144a5010ee435ac5b207a7c4b89d15fd
<pre>
git-webkit is not posting PR links to bugzilla bugs
<a href="https://rdar.apple.com/145594374">rdar://145594374</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=288151">https://bugs.webkit.org/show_bug.cgi?id=288151</a>

Reviewed by Sam Sneddon.

Check whether the first issue is a bugzilla bug.
If not, explicitly comment on the bugzilla instead.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.add_comment_to_issue):
(PullRequest.create_pull_request):

Canonical link: <a href="https://commits.webkit.org/292453@main">https://commits.webkit.org/292453@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1b9321d892f07952ea0fec4a0b4720c34f85f50a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96103 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/15717 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/5673 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101166 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/46612 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/98148 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16012 "Build is in progress. Recent messages:OS: Sequoia (15.2), Xcode: 16.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24150 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/73258 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30478 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99106 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/11995 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/86809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53596 "Passed tests") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/95590 "Found 2 webkitpy test failures: webkit.messages_unittest.GeneratedFileContentsTest.test_message_argument_description, webkit.messages_unittest.GeneratedFileContentsTest.test_message_names") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/11744 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/4565 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45947 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/81878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/4666 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103193 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23170 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/82296 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/95674 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/23421 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82825 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81669 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26283 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/3715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/16526 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15466 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23133 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/22792 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/26272 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/24533 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
```diff
@@ -697,18 +697,24 @@ def create_pull_request(cls, repository, args, branch_point, callback=None, unbl
-                else:
-                    issue.open(why='Re-opening for pull request {}'.format(pr.url))
-                print('Posted pull request link to {}'.format(issue.link))
+        def comment_on_issue(issue):
```
I feel like this may as well be a method on the class in its own right, because that only means additionally taking `pr` as an argument, and then there's slightly less complexity in what's currently a 333 line function?

gsnedders commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
```diff
@@ -404,6 +404,21 @@ def add_comment_to_reverted_commit_bug_tracker(cls, repository, args, pr, commit
 
+    @classmethod
+    def add_comment_to_issue(cls, issue, update_issue, pr):
+        if issue and update_issue:
```
I feel like having a function both take these arguments (especially update_issue, which I believe to be a bool) and then immediately have an `if` branch is kinda surprising.

Like, as I understand it, the interface is currently:

```python
def add_comment_to_issue(cls, issue: Tracker | None, update_issue: bool, pr: PullRequest) -> None: ...
```

Whereas I think I'd expect:

```python
def add_comment_to_issue(cls, issue: Tracker, pr: PullRequest) -> None: ...
```

And then it's up the caller to ensure it's not None, and to only call it when the update is wanted?

briannafan replied:
Great point - I think this weird interface was due to me not thinking and just copy-pasting existing code into a function.

briannafan committed:
git-webkit is not posting PR links to bugzilla bugs
rdar://145594374
https://bugs.webkit.org/show_bug.cgi?id=288151

Reviewed by Sam Sneddon.

Check whether the first issue is a bugzilla bug.
If not, explicitly comment on the bugzilla instead.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.add_comment_to_issue):
(PullRequest.create_pull_request):

Canonical link: https://commits.webkit.org/292453@main
